File: packages/transform/src/features/protobuf/ProtobufValidateDecodeTransformer.ts
import { ProtobufValidateDecodeProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufValidateDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.validateDecode",
      write: ProtobufValidateDecodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufValidateDecodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufValidateDecodeTransformer = protobufValidateDecodeTransformerNamespace{}

type protobufValidateDecodeTransformerNamespace struct{}

func (protobufValidateDecodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufValidateDecodeProgrammer.Write(call.TypeText)
}
